test/Elastic.Apm.Feature.Tests/CloudProviderSteps.cs

```diff
@@ -0,0 +1,152 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Report;
+using Elastic.Apm.Tests.Utilities;
+using Elastic.Apm.Tests.Utilities.XUnit;
+using FluentAssertions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RichardSzalay.MockHttp;
+using TechTalk.SpecFlow;
+using Xunit.Abstractions;
+using static Elastic.Apm.BackendComm.BackendCommUtils.ApmServerEndpoints;
+using MockHttpMessageHandler = RichardSzalay.MockHttp.MockHttpMessageHandler;
+
+namespace Elastic.Apm.Feature.Tests
+{
+	[Binding]
+	public class CloudProviderSteps
+	{
+		private static readonly JsonSerializer _serializer = new JsonSerializer();

```

Comment:
Shouldn't we reuse `PayloadItemSerializer` here? Just to be in-sync with the one we use in production in `PayloadSenderV2`.

Reply:
It doesn't make a difference; the assertions are against JSON, and parsed into a `JObject` to make traversing easier